src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -926,16 +926,22 @@ private ISwitchStatement CreateBoundSwitchStatementOperation(BoundSwitchStatemen
             return new LazySwitchStatement(value, cases, isInvalid, syntax, type, constantValue);
         }
 
-        private ISingleValueCaseClause CreateBoundSwitchLabelOperation(BoundSwitchLabel boundSwitchLabel)
+        private ICaseClause CreateBoundSwitchLabelOperation(BoundSwitchLabel boundSwitchLabel)
         {
-            Lazy<IOperation> value = new Lazy<IOperation>(() => Create(boundSwitchLabel.ExpressionOpt));
-            BinaryOperationKind equality = GetLabelEqualityKind(boundSwitchLabel);
-            CaseKind caseKind = boundSwitchLabel.ExpressionOpt != null ? CaseKind.SingleValue : CaseKind.Default;
             bool isInvalid = boundSwitchLabel.HasErrors;
             SyntaxNode syntax = boundSwitchLabel.Syntax;
             ITypeSymbol type = null;
             Optional<object> constantValue = default(Optional<object>);
-            return new LazySingleValueCaseClause(value, equality, caseKind, isInvalid, syntax, type, constantValue);
+            if (boundSwitchLabel.ExpressionOpt != null)
+            {
+                Lazy<IOperation> value = new Lazy<IOperation>(() => Create(boundSwitchLabel.ExpressionOpt));
+                BinaryOperationKind equality = GetLabelEqualityKind(boundSwitchLabel);
+                return new LazySingleValueCaseClause(value, equality, CaseKind.SingleValue, isInvalid, syntax, type, constantValue);
+            }
+            else
```

Comment:
@gafter Is it correct to assume that any BoundSwitchLabel with no ExpressionOpt must be the "default" label?